lib/torrent.js

```diff
@@ -36,6 +36,43 @@ import { Selections } from './selections.js'
 
 import VERSION from '../version.cjs'
 
+// The following JSDoc comments are global types that can be used across the project
+
+/**
+ * This callback is called with an optional error, if the error is a falsy value the operation was executed successfully
+ * @callback callbackWithError
+ * @param {Error=} error
+ */
+
+/**
+ * @typedef TorrentOpts
+ * @type {object}
+ * @property {Array<string>=} announce - Torrent trackers to use (added to list in .torrent or magnet uri)
+ * @property {Array<string>=} urlList - Array of web seeds
+ * @property {string=} path - Folder to download files to (default=`/tmp/webtorrent/`)
+ * @property {boolean=} addUID - (Node.js only) If true, the torrent will be stored in it's infoHash folder to prevent file name collisions (default=false)
+ * @property {FileSystemDirectoryHandle=} rootDir - *(browser only)* if supported by the browser, allows the user to specify a custom directory to stores the files in, retaining the torrent's folder and file structure
+ * @property {boolean=} skipVerify - If true, client will skip verification of pieces for existing store and assume it's correct
+ * @property {Uint8Array|Buffer|Array<number>=} bitfield -  Preloaded numerical array/buffer to use to know what pieces are already downloaded (any type accepted by UInt8Array constructor is valid)
```

Comment:
Uint8Array, not Buffer, never rely on buffer if possible here, because wt runs in browser, which doesnt have buffer

Reply:
However Node has it, so this type has some types for the browser and some for Node, ideally it should overlap with the types allowed in `new UInt8Array(<<<here>>>)`

Reply:
This is from `lib.es5.d.ts` ( Typescript internals )

```
    new (length: number): Uint8Array;
    new (array: ArrayLike<number> | ArrayBufferLike): Uint8Array;
    new (buffer: ArrayBufferLike, byteOffset?: number, length?: number): Uint8Array;
```

I'm sure we don't want it to be a simple number cause it would make no sense ( even if it would work ), so the correct types are:

`ArrayLike<number` and `ArrayBufferLike`

```ts
interface ArrayLike<T> {
    readonly length: number;
    readonly [n: number]: T;
}

interface ArrayBufferTypes {
    ArrayBuffer: ArrayBuffer;
}
type ArrayBufferLike = ArrayBufferTypes[keyof ArrayBufferTypes];
```

Reply:
I changed `Buffer` to `ArrayBuffer` cause it's a superset

Reply:
Buffer is Uint8Array with some extra functions, buffer extends uint8array, which extends typedarray, arraybuffer is a representation of data that cannot be accessed without a view, buffer doesnt exist in browsers so as a rule of thumb uint8array is used everywhere in webtorrent